src/server/aws.ts

```diff
@@ -20,20 +19,24 @@ export function objectToAWSTags(tags: Record<string, string>) {
     }))
 }
 
-// TODO Remove ecrCLient unused?
-let _ecrClient: ECRClient | null = null
-export const getECRClient = () =>
-    _ecrClient ||
-    (_ecrClient = new ECRClient({
-        region: process.env.AWS_REGION || 'us-east-1',
-        credentials: process.env.AWS_PROFILE ? fromIni({ profile: process.env.AWS_PROFILE }) : undefined,
-    }))
-
 let _s3Client: S3Client | null = null
 export const getS3Client = () =>
     _s3Client ||
     (_s3Client = new S3Client({
         region: process.env.AWS_REGION || 'us-east-1',
+        forcePathStyle: true,
+        endpoint: process.env.S3_ENDPOINT,
+        credentials: process.env.AWS_PROFILE ? fromIni({ profile: process.env.AWS_PROFILE }) : undefined,
+    }))
+
+// For Presigned URLs and client calls
+let _s3BrowserClient: S3Client | null = null
+export const getS3BrowserClient = () =>
+    _s3BrowserClient ||
+    (_s3BrowserClient = new S3Client({
+        region: process.env.AWS_REGION || 'us-east-1',
+        forcePathStyle: true,
+        endpoint: process.env.S3_BROWSER_ENDPOINT || process.env.S3_ENDPOINT,
         credentials: process.env.AWS_PROFILE ? fromIni({ profile: process.env.AWS_PROFILE }) : undefined,
```

Comment:
I don't fully understand the ramifications of using 2 s3Clients or what the `forcePathStyle: true,` does, it seem harmless though so as long as it works I'm ok with it

Reply:
yeah the main motivation for doing this was something I believe we saw when testing on a call together, when we tried to change the URL. It will reject it if the URL is different than what it's originally signed with, and the URL comes from the client so it made sense to me when thinking about it that way 🤣